packages/xstate-react/src/useReactEffectActions.ts

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useRef } from 'react';
+import useIsomorphicLayoutEffect from 'use-isomorphic-layout-effect';
+import { EventObject, State, Interpreter } from 'xstate';
+import { ReactActionObject, ReactEffectType } from './types';
+import { partition } from './utils';
+import { ActionStateTuple, executeEffect } from './useMachine';
+
+export function useReactEffectActions<TContext, TEvent extends EventObject>(
+  service: Interpreter<TContext, any, TEvent, any>
+) {
+  const effectActionsRef = useRef<
+    Array<
+      [ReactActionObject<TContext, TEvent>, State<TContext, TEvent, any, any>]
+    >
+  >([]);
+  const layoutEffectActionsRef = useRef<
+    Array<
+      [ReactActionObject<TContext, TEvent>, State<TContext, TEvent, any, any>]
+    >
+  >([]);
+
+  useIsomorphicLayoutEffect(() => {
+    const sub = service.subscribe((currentState) => {
```

Comment:
Because of the implemented split of things, some initial React actions will be missed - you could end up with such a situation even more easily than in the test case provided here:
https://github.com/davidkpiano/xstate/pull/1922/files#diff-753cb411217f73964488b6bea771171aaef83d7e7f66c3717d5ae3103f90bdb3R733

It would only require:
- having an initial React action
- sending an event within initial actions

This is an edge case so I don't think there is much need to fix this right now but we probably should think about how we could fix this in the future.

Random thought: after more thinking about "React actions" I'm starting to think that this is really quirky and maybe we should revisit this in the future altogether. React's model is "synchronize to state" and maybe this is just what we should stick to in this regard.